flask_multipass_cern.py

```diff
@@ -98,10 +113,21 @@ def get_members(self):
     def has_member(self, identifier):
         cache_key = f'flask-multipass-cern:{self.provider.name}:groups:{identifier}'
         all_groups = self.provider.cache and self.provider.cache.get(cache_key)
-        if all_groups is None:
-            all_groups = {g.name.lower() for g in self.provider.get_identity_groups(identifier)}
-            if self.provider.cache:
-                self.provider.cache.set(cache_key, all_groups, 1800)
+        should_refresh = self.provider.cache and self.provider.cache.get(f'{cache_key}:timestamp') is None
+
+        if all_groups is None or should_refresh:
+            try:
+                all_groups = {g.name.lower() for g in self.provider.get_identity_groups(identifier)}
+                if self.provider.cache:
+                    self.provider.cache.set(cache_key, all_groups, timeout=CACHE_LONG_TTL)
+                    self.provider.cache.set(f'{cache_key}:timestamp', datetime.now(), timeout=CACHE_TTL)
+            except HTTPError:
+                logger = logging.getLogger(self.provider.settings['logger_name'])
```

Comment:
I'd assign the logger as `self.logger` in the constructor since we need it in multiple places